winpr/libwinpr/utils/collections/HashTable.c

```diff
@@ -74,24 +105,24 @@ void HashTable_StringFree(void* str)
 	free(str);
 }
 
-static int HashTable_IsProbablePrime(int oddNumber)
+static BOOL HashTable_IsProbablePrime(size_t oddNumber)
 {
-	int i;
+	size_t i;
 
 	for (i = 3; i < 51; i += 2)
 	{
 		if (oddNumber == i)
-			return 1;
+			return TRUE;
 		else if (oddNumber % i == 0)
-			return 0;
+			return FALSE;
 	}
 
-	return 1; /* maybe */
+	return TRUE; /* maybe */
 }
 
-static long HashTable_CalculateIdealNumOfBuckets(wHashTable* table)
+static size_t HashTable_CalculateIdealNumOfBuckets(wHashTable* table)
```

Comment:
Perhaps we can have a better way to compute ideal sizes as it's done here: https://github.com/microsoft/weston-mirror/blob/working/libweston/backend-rdp/hash.c#L67 (no computations, just a table scan)